Panic when checking info nodes

The flask demo crashes qmstr-master.

`2019/11/19 13:15:47 Running analyzer scancode-analyzer ...
2019/11/19 13:15:47 Opened insert queue
2019/11/19 13:17:48 Already created info nodes for node 0x2a, skipping insert..
panic: runtime error: index out of range

goroutine 9448 [running]:
github.com/QMSTR/qmstr/lib/go-qmstr/database.AnalyzerCreatedInfoNodes(0x90c080, 0xc000538e20, 0x907e00, 0xc000329b88, 0xc000428e28)
	/qmstr/lib/go-qmstr/database/infoNode.go:100 +0x410
github.com/QMSTR/qmstr/lib/go-qmstr/database.(*DataBase).AddInfoNodes(0xc0001e4030, 0xc000027fd8, 0x4, 0x0, 0x0, 0x0, 0x0, 0x0)
	/qmstr/lib/go-qmstr/database/infoNode.go:54 +0x87c
github.com/QMSTR/qmstr/lib/go-qmstr/master.(*serverPhaseAnalysis).SendInfoNodes(0xc00029c910, 0xab2c40, 0xc0001284b0, 0x1, 0xc0001284b0)
	/qmstr/lib/go-qmstr/master/analyze.go:126 +0x108
github.com/QMSTR/qmstr/lib/go-qmstr/master.(*server).SendInfoNodes(0xc000020e00, 0xab2c40, 0xc0001284b0, 0xaaeaa0, 0xc000020e00)
	/qmstr/lib/go-qmstr/master/master.go:77 +0x4b
github.com/QMSTR/qmstr/lib/go-qmstr/service._AnalysisService_SendInfoNodes_Handler(0x9d7a00, 0xc000020e00, 0xab1260, 0xc000454180, 0xeb15c0, 0xc00030c000)
	/qmstr/lib/go-qmstr/service/analyzerservice.pb.go:539 +0xad
google.golang.org/grpc.(*Server).processStreamingRPC(0xc00009af20, 0xab3480, 0xc0003ae600, 0xc00030c000, 0xc0000fec90, 0xe89840, 0x0, 0x0, 0x0)
	/root/go/pkg/mod/google.golang.org/grpc@v1.23.0/server.go:1199 +0xa74
google.golang.org/grpc.(*Server).handleStream(0xc00009af20, 0xab3480, 0xc0003ae600, 0xc00030c000, 0x0)
	/root/go/pkg/mod/google.golang.org/grpc@v1.23.0/server.go:1279 +0xd3f
google.golang.org/grpc.(*Server).serveStreams.func1.1(0xc0003f62f0, 0xc00009af20, 0xab3480, 0xc0003ae600, 0xc00030c000)
	/root/go/pkg/mod/google.golang.org/grpc@v1.23.0/server.go:710 +0x9f
created by google.golang.org/grpc.(*Server).serveStreams.func1
	/root/go/pkg/mod/google.golang.org/grpc@v1.23.0/server.go:708 +0xa1`

--- a/modules/analyzers/scancode-analyzer/main.go
+++ b/modules/analyzers/scancode-analyzer/main.go
@@ -95,18 +95,20 @@ func (scanalyzer *ScancodeAnalyzer) Analyze(controlService service.ControlServic
 			infoNodes = append(infoNodes, copyrights)
 		}
 
-		infoNodeMsgs = append(infoNodeMsgs, &service.InfoNodesMessage{Token: token, Infonodes: infoNodes, Uid: fileNode.FileData.Uid})
+		if len(infoNodes) > 0 {
+			infoNodeMsgs = append(infoNodeMsgs, &service.InfoNodesMessage{Token: token, Infonodes: infoNodes, Uid: fileNode.FileData.Uid})
+		}
 	}
 
-	send_stream, err := analysisService.SendInfoNodes(context.Background())
+	sendStream, err := analysisService.SendInfoNodes(context.Background())
 	if err != nil {
 		return err
 	}
 	for _, inodeMsg := range infoNodeMsgs {
-		send_stream.Send(inodeMsg)
+		sendStream.Send(inodeMsg)
 	}
 
-	reply, err := send_stream.CloseAndRecv()
+	reply, err := sendStream.CloseAndRecv()
 	if err != nil {
 		return err
 	}

